integration_tests/structs_22.py

```diff
@@ -0,0 +1,44 @@
+from lpython import dataclass, i32, f64, u64
+from numpy import array
+
+@dataclass
+class Foo:
+    x: i32
+    y: i32
+
+@dataclass
+class Foo2:
+    p: f64
+    q: i32
+    r: u64
+
+def main0() -> None:
+    foos: Foo[2] = array([Foo(1, 2), Foo(3, 4)])
+    print(foos[0].x, foos[0].y, foos[1].x, foos[1].y)
+
+    assert foos[0].x == 1
+    assert foos[0].y == 2
+    assert foos[1].x == 3
+    assert foos[1].y == 4
+
+def main1() -> None:
+    foos2: Foo2[3] = array([])
```

Comment:
The RHS is not correct, it's an array of length 0, assigning to array of length 3. So this should give either runtime or compile time error.